### PR TITLE
cgroup2: implement `docker stats`

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**

Implemented `docker stats`  for cgroup2.

The following fields are unsupported (not used by `docker` CLI):
    
* `BlkioStats`: all fields other than `IoServiceBytesRecursive`
* `CPUStats`: `CPUUsage.PercpuUsage`
* `MemoryStats`: `MaxUsage` and `Failcnt`

**- How I did it**

See the code.

**- How to verify it**
```console
$ docker run -d --name foo --cpus 0.5 --memory 32m --runtime crun alpine \
  sh -exc "apk add --no-cache stress-ng && stress-ng --cpu 0"
f3085a21a2e2f0aa1e6d3fc8682aa47f233c561d9f8aced9fd2dfd6297f742e3

$ docker stats
CONTAINER ID        NAME                CPU %               MEM USAGE / LIMIT   MEM %               NET I/O             BLOCK I/O           PIDS
afaae62989ba        foo                 49.92%              13.19MiB / 32MiB    41.22%              4.07MB / 42.7kB     0B / 3.48MB         3
```

Tested on Ubuntu 19.10, containerd/containerd@936b7b0043e4b90a7f4810ed91ce066257117178, crun v0.13.

**- Description for the changelog**

cgroup2: implement `docker stats`

**- A picture of a cute animal (not mandatory but encouraged)**
:penguin:

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
@Zyqsempai Are you interested in filling up missing `containerd/cgroups` fields?

**Comment by Zyqsempai** [comment]
@AkihiroSuda Yes I am, thanks!

**Comment by AkihiroSuda** [comment]
Added blkio stats. Should be ready to review and merge.

File: libcontainerd/types/types_linux.go [hunk 1]
```diff
@@ -12,14 +11,17 @@ type Summary struct{}
 
 // Stats holds metrics properties as returned by containerd
 type Stats struct {
-	Read    time.Time
-	Metrics *statsV1.Metrics
+	Read time.Time
+	// Metrics is expected to be either one of:
+	// * github.com/containerd/cgroups/stats/v1.Metrics
+	// * github.com/containerd/cgroups/stats/v2.Metrics
+	Metrics interface{}
```
Comment (cpuguy83) [inline libcontainerd/types/types_linux.go]: Could this be 2 separate fields instead?

Comment (AkihiroSuda) [reply libcontainerd/types/types_linux.go]: I think `interface{}` is better for extensibility. Potentially non-runc runtime may produce other type of Metrics.

Comment (cpuguy83) [reply libcontainerd/types/types_linux.go]: How would we consume that?

Comment (AkihiroSuda) [reply libcontainerd/types/types_linux.go]: We would need  to add `switch Metrics.(type)` cases as in this PR.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by AkihiroSuda** [comment]
@thaJeztah PTAL?

**Comment by AkihiroSuda** [comment]
@kolyshkin PTAL?

**Review by kolyshkin — APPROVED** [review]
SGTM

**Comment by AkihiroSuda** [comment]
Can we merge this?

**Comment by thaJeztah** [comment]
IIUC, some information will be different / not present depending on if V1 or V2 is used? If so, perhaps we should add a note to the swagger docs to mention this?

I'm ok if that's done in a follow-up

**Comment by AkihiroSuda** [comment]
added swagger doc

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**